### ISSUE TITLE
Bundle error when loading `realm@latest` with `react-native@0.69.0-rc.0`: "dependency.platforms.ios.podspecPath" is not allowed

### ISSUE THREAD

**Issue by kraenhansen** [realm/realm-js#4553]
### How frequently does the bug occur?

Sometimes for end-users, as they have to be using the specific versions.

### Description

The React Native install tests has [started failing](https://github.com/realm/realm-js/runs/6269911169?check_suite_focus=true) due to.

### Stacktrace & log output

```shell
Warning: Package realm has been ignored because it contains invalid configuration.
Reason: "dependency.platforms.ios.podspecPath" is not allowed

// Followed by

Error: Missing Realm constructor. Did you run "pod install"? Please see https://realm.io/docs/react-native/latest/#missing-realm-constructor for troubleshooting
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

1. npm install realm@10.16.0 react-native@0.69.0-rc.0
2. Try to require the `realm` package.
3. See the warning and error in the terminal running metro.

### Version

10.16.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

React Native iOS

### Build environment

Which debugger for React Native: N/A

**Comment by kraenhansen** [comment]
This got removed recently: [https://github.com/react-native-community/cli/commit/25eec7c695f09aea0ace7c0b591844fe8828ccc5](https://github.com/react-native-community/cli/commit/25eec7c695f09aea0ace7c0b591844fe8828ccc5)

**Comment by mordechaim** [comment]
This isn't merged with the hermes beta version (10.20.0-beta.5) yet.

`realm+10.20.0-beta.5.patch`
```diff
diff --git a/node_modules/realm/react-native.config.js b/node_modules/realm/react-native.config.js
index 84b0ce0..79ebe6e 100644
--- a/node_modules/realm/react-native.config.js
+++ b/node_modules/realm/react-native.config.js
@@ -27,10 +27,7 @@ module.exports = {
       android: {
         sourceDir: "./react-native/android",
       },
-      ios: {
-        podspecPath: path.resolve(__dirname, "RealmJS.podspec"),
-        sharedLibraries: ["libc++", "libz"],
-      },
+      ios: {},
     },
   },
 };
```
